ui/app/AppLayouts/Wallet/services/dapps/DAppsListProvider.qml

```diff
@@ -52,24 +68,34 @@ QObject {
                 sdk.getActiveSessions((allSessions) => {
                     root.store.dappsListReceived.disconnect(dappsListReceivedFn);
 
-                    let tmpMap = {}
-                    var topics = []
-                    const sessions = DAppsHelpers.filterActiveSessionsForKnownAccounts(allSessions, root.supportedAccountsModel)
-                    for (const key in sessions) {
-                        const dapp = sessions[key].peer.metadata
+                    const dAppsMap = []
+                    const topics = []
+                    for (const sessionID in sessions) {
+                        const session = sessions[sessionID]
+                        const dapp = session.peer.metadata
                         if (!!dapp.icons && dapp.icons.length > 0) {
                             dapp.iconUrl = dapp.icons[0]
                         } else {
                             dapp.iconUrl = ""
                         }
-                        tmpMap[dapp.url] = dapp;
-                        topics.push(key)
+                        const accounts = DAppsHelpers.getAccountsInSession(session)
+                        const existingDApp = dAppsMap[dapp.url]
+                        if (existingDApp) {
+                            existingDApp.accountAddresses = [...new Set([...existingDApp.accountAddresses, ...dapp.accountAddresses])];
+                        } else {
+                            dapp.accountAddresses = accounts
+                            dAppsMap[dapp.url] = dapp
+                        }
+
+                        topics.push(sessionID)
                     }
+
                     // TODO #15075: on SDK dApps refresh update the model that has data source from persistence instead of using reset
                     dapps.clear();
-                    // Iterate tmpMap and fill dapps
-                    for (const key in tmpMap) {
-                        dapps.append(tmpMap[key]);
+
+                    // Iterate dAppsMap and fill dapps
+                    for (const topic in dAppsMap) {
+                        dapps.append(dAppsMap[topic]);
```

Comment:
It's good practice to append array in one call whenever possible.
E.g. sth like that (not tested):
```
dapps.append(dAppsMap.map(t => dAppsMap[t]))
```

Reply:
Btw, ideally at this stage we should more carefully inspect the incoming data before putting into the model. We can take only what's needed and sanitize (e.g. if some fields are optional and may be not present we should provide e.g. "" instead in order to have always the same set of roles). Then `dynamicRoles` and tricks with `SFPM` won't be needed at all.